### PR TITLE
MES-1421: Use staffNumber as path parameter in request to get journal

### PR DESCRIPTION
**Pull request by johnmorleybjss**
## Description and relevant Jira numbers
In order to request an individual examiner's journal, we need to pass the `staffNumber` as a path parameter. Retrieve the cached staff number from the `AuthenticationProvider` and use it to populate the URL template from config.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/providers/app-config/__mocks__/app-config.mock.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ export class AppConfigProviderMock {
     return;
   }
 
+  public getPersonalJournalUrl(): string {
+    return 'https://www.example.com/api/v1/journals/1234/personal'
+  }
```
Comment (Cavalier07) [inline src/providers/app-config/__mocks__/app-config.mock.ts]: Could we make the mock id 8 digits so it matches the data we have in Active Directory

File: src/providers/journal/__tests__/journal.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    let appConfig;
+    
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [
+          HttpClientTestingModule
+        ],
+        providers: [
+          JournalProvider,
+          { provide: AppConfigProvider, useClass: AppConfigProvider, environmentFile: remoteEnvironmentMock },
```
Comment (Cavalier07) [inline src/providers/journal/__tests__/journal.spec.ts]: Should we not be using the mock here for the AppConfigProvider.

You then don't have to mock the employee id later on in the test.

File: src/providers/app-config/app-config.ts [hunk 3]
```diff
@@ -29,6 +29,10 @@ export class AppConfigProvider {
     return this.appConfig;
   }
 
+  public getPersonalJournalUrl(staffNumber: string) {
+    return this.appConfig.journal.journalUrl.replace('{staffNumber}', staffNumber || '1');
```
Comment (szabolcs-keresztes) [inline src/providers/app-config/app-config.ts]: Why are we defaulting to `1`?

**Comment by szabolcs-keresztes** [comment]
Good work! Just one small thing came to my mind, you'll see in the comments

File: src/providers/app-config/__mocks__/app-config.mock.ts (hunk 1 above)
Comment (johnmorleybjss) [reply src/providers/app-config/__mocks__/app-config.mock.ts]: Done.

File: src/providers/app-config/app-config.ts (hunk 3 above)
Comment (johnmorleybjss) [reply src/providers/app-config/app-config.ts]: Defaulted to `00000000`.

**Pull request merged**